server/src/main/java/org/candlepin/guice/CandlepinModule.java

```diff
@@ -424,9 +449,33 @@ private void configureEventSink() {
     }
 
     protected void configureModelTranslator() {
-        bind(APIModelTranslator.class).to(APIv1ModelTranslator.class);
-        // bind(APIv1ModelTranslator.class).to(APIv1ModelTranslator.class);
-
-        // TODO: Add more factory configurations here as necessary
+        ModelTranslator modelTranslator = new SimpleModelTranslator();
+
+        // API translators
+        /////////////////////////////////////////////
+        modelTranslator.registerTranslator(
+            new CertificateSerialTranslator(), CertificateSerial.class, CertificateSerialDTO.class);
+        modelTranslator.registerTranslator(
+            new CertificateTranslator(), Certificate.class, CertificateDTO.class);
+        modelTranslator.registerTranslator(
+            new ConsumerTypeTranslator(), ConsumerType.class, ConsumerTypeDTO.class);
+        modelTranslator.registerTranslator(
+            new ContentTranslator(), Content.class, ContentDTO.class);
+        modelTranslator.registerTranslator(
+            new OwnerTranslator(), Owner.class, OwnerDTO.class);
+        modelTranslator.registerTranslator(
+            new ProductTranslator(), Product.class, ProductDTO.class);
+        modelTranslator.registerTranslator(
+            new UpstreamConsumerTranslator(), UpstreamConsumer.class, UpstreamConsumerDTO.class);
+
+        // Shims
+        /////////////////////////////////////////////
```

Comment:
Perhaps add a comment here detailing why the shims are required.

Reply:
I could write a whole book on it, but I added a quick two-liner.